pinot-core/src/main/java/org/apache/pinot/core/query/scheduler/QueryScheduler.java

```diff
@@ -306,6 +306,10 @@ private boolean forceLog(long schedulerWaitMs, long numDocsScanned) {
   protected ListenableFuture<byte[]> immediateErrorResponse(ServerQueryRequest queryRequest,
       ProcessingException error) {
     DataTable result = new DataTableImplV2();
+
+    Map<String, String> dataTableMetadata = result.getMetadata();
```

Comment:
Can we consider making this DataTable a static one that is reused across all failing queries, to avoid creating new objects? Or is there a case where that doesn't work?

Reply:
From what I can tell, using a static DataTable object will lead to race conditions because different threads will try to set their own request ids into the static DataTable.

Reply:
Yeah, we might want to set different exceptions as well.